tests/integration/test_orders_api.py

```diff
@@ -645,6 +645,37 @@ async def test_download_order_state(tmpdir, order_description, oid, session):
         await cl.download_order(oid, directory=str(tmpdir))
 
 
+@respx.mock
+@pytest.mark.asyncio
+async def test_download_order_without_order_links(tmpdir,
+                                                  order_description,
+                                                  oid,
+                                                  session):
+    dl_url1 = TEST_DOWNLOAD_URL + '/1?token=IAmAToken'
+
+    order_description['state'] = 'success'
+    order_description['_links']['results'] = None
```

Comment:
Just a conversation: It is kind of wacky that the order is in a success state and yet has no results to download. I wonder if we ever actually see that scenario.

Reply:
Right, totally agree, but for some reason that bubbled up in #325, so maybe just a weird edge case?